docs/development/local_setup.md

```diff
@@ -69,31 +105,71 @@ $ git clone git@github.com:gardener/gardener.git
 $ cd gardener
 ```
 
-# Local development
+## Start the Gardener
 
 :warning: Before you start developing, please ensure to comply with the following requirements:
 
 1. You have understood the [principles of Kubernetes](https://kubernetes.io/docs/concepts/), and its [components](https://kubernetes.io/docs/concepts/overview/components/), what their purpose is and how they interact with each other.
 1. You have understood the [architecture of the Garden](https://github.com/gardener/documentation/wiki/Architecture), and what the various clusters are used for.
 
-The development of the Gardener could happen by targetting any cluster. You basically need a Garden cluster (e.g., a [Minikube](https://github.com/kubernetes/minikube) cluster) and one Seed cluster per cloud provider and per data center/region. You can configure the Gardener controller manager to watch **all namespaces** for Shoot manifests or to only watch **one single** namespace.
+The development of the Gardener could happen by targeting any cluster. You basically need a Garden cluster (e.g., a [Minikube](https://github.com/kubernetes/minikube) cluster) and one Seed cluster per cloud provider and per data center/region. You can configure the Gardener controller manager to watch **all namespaces** for Shoot manifests or to only watch **one single** namespace.
 
-In order to ensure that a specific Seed cluster will be chosen, add the `.spec.cloud.seed` field (see [here](../../example/shoot-azure.yaml#L10) for an example Shoot manifest).
+The commands bellow will set your `minikube` with the absolute minimum resources to launch Gardener API server & controller manager:
 
-## Get started
+```bash
+$ minikube start
+Starting local Kubernetes v1.8.0 cluster...
+[...]
+kubectl is now configured to use the kluster.
+
```

Comment:
Can we add a section here and mention that, before one can execute `make dev-setup`, it is required to create at least the internal domain secret based on the manifest in the `example` folder? We could link to this documentation page: https://github.com/gardener/gardener/blob/master/docs/deployment/configuration.md
I have something like that in mind:

```bash
$ minikube start
Starting local Kubernetes v1.8.0 cluster...
[...]
```

The Gardener exposes the API servers of Shoot clusters via Kubernetes services of type `LoadBalancer`. In order to establish stable endpoints (robust against changes of the load balancer address), it creates DNS records pointing to these load balancer addresses. They are used internally and by all cluster components to communicate.
You need to have control over a domain (or subdomain) for which these records will be created.
Please provide an *internal domain secret* (see [this](https://github.com/gardener/gardener/blob/master/example/secret-internal-domain.yaml) for an example) which contains credentials with the proper privileges. Further information can be found [here](https://github.com/gardener/gardener/blob/master/docs/deployment/configuration.md).

```bash
$ mkdir -p dev
$ cp example/secret-internal-domain.yaml dev/secret-internal-domain.yaml
<Put your credentials in dev/secret-internal-domain.yaml>
$ kubectl apply -f dev/secret-internal-domain.yaml
secret "internal-domain-example-com" created

$ make dev-setup
...
```

What do you think? If you want to mention default domain secrets, please do so.

Reply:
I was thinking more in the way of something in the middle

1. The example internal domain has a dummy configuration:
   ```yaml
   data:
     accessKeyID: dGVzdAo= # base64(access-key-id-with-route53-privileges)
   ```
1. We copy it to `dev` and apply it from there.
1. The end user then can modify it and apply itself if needed.

Reply:
Hm, what do you mean by "if needed"? It will always be necessary to change that (except that somebody only wants to develop against Vagrant which I assume is rather the minority).
If we would apply this "dummy configuration" the initial setup won't work until the user applies the modified internal domain secret itself and restarts the controller manager (both additional complexity and documentation effort for an initial setup).

The standard development case (at least at the moment) is to develop against cloud providers, i.e. a real internal domain secret is required. Thus, the "default/standard local dev setup" should have the required steps to establish that scenario easily and with least complexity.